src/EA.Weee.Web/Areas/Admin/Controllers/ProducerSubmissionController.cs

```diff
@@ -129,16 +134,87 @@ public async Task<ActionResult> AddPaymentDetails()
             return View();
         }
 
+        [AdminSmallProducerSubmissionContext]
         [HttpGet]
-        public async Task<ActionResult> RemoveSubmission()
+        public async Task<ActionResult> RemoveSubmission(string registrationNumber, int year)
         {
-            return View();
+            var submission = this.SmallProducerSubmissionData.SubmissionHistory[year];
+            var selectedValue = string.Empty;
+            var model = new ConfirmRemovalViewModel
+            {
+                SelectedValue = selectedValue,
+                Producer = new ProducerDetailsScheme
+                {
+                    ComplianceYear = submission.ComplianceYear,
+                    ProducerName = submission.CompanyName,
+                    RegistrationNumber = registrationNumber,
+                    RegisteredProducerId = submission.RegisteredProducerId
+                }
+            };
+
+            return View(model);
+        }
+        
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> RemoveSubmission(ConfirmRemovalViewModel viewModel)
+        {
+            if (this.ModelState.IsValid)
+            {
+                if (viewModel.SelectedValue == "Yes")
+                {
+                    RemoveSmallProducerResult result;
+                    using (IWeeeClient client = apiClient())
+                    {
+                        result = await client.SendAsync(User.GetAccessToken(), new RemoveSmallProducer(viewModel.Producer.RegisteredProducerId));
+                    }
+
+                    if (result.InvalidateProducerSearchCache)
+                    {
+                        await cache.InvalidateProducerSearch();
+                    }
+
+                    return RedirectToAction(nameof(ProducerSubmissionController.Removed),
+                        new { producerName = viewModel.Producer.ProducerName, year = viewModel.Producer.ComplianceYear });
+                }
+                else
+                {
+                    return RedirectToAction(nameof(ProducerSubmissionController.Submissions),
+                        new { viewModel.Producer.RegistrationNumber });
+                }
+            }
+            else
+            {
+                return this.View(viewModel);
+            }
         }
 
         [HttpGet]
-        public async Task<ActionResult> ReturnProducerRegistration()
+        public async Task<ActionResult> Removed(string producerName, int year)
```

Comment:
@stevelee-civica doesnt need to be async.

Reply:
updated